### ISSUE TITLE
overfit to context views

### ISSUE THREAD

**Issue by LiuChunlin-eng** [OpenRobotLab/AnySplat#89]
Sorry to bother you again. I tried to go through the training process on the dl3dv dataset. When training on the dl3dv dataset, overfitting to the top and bottom views is not obvious or common. However, when I switch to a custom dataset and train for fewer steps, I notice the views are as follows:
<img width="940" height="715" alt="Image" src="https://github.com/user-attachments/assets/388f2533-0e3f-4a4b-8e13-70f9b63c79cf" />
I want to ask:
1. Is this phenomenon normal?
2. My next step is to try loading the anysplat weights or modifying the data loading sampling strategy. Do you have any good suggestions?

**Comment by Alex417ii** [comment]
Hello, I am currently attempting to train on a small DL3DV dataset, but I have encountered instability during the training process. May I ask if you have experienced this situation during your training? I was wondering if you are using the original DL3DV dataset, and if there have been any modifications to the parameter configurations or code sections? I look forward to your reply and sincerely appreciate your help.

> Sorry to bother you again. I tried to go through the training process on the dl3dv dataset. When training on the dl3dv dataset, overfitting to the top and bottom views is not obvious or common. However, when I switch to a custom dataset and train for fewer steps, I notice the views are as follows:抱歉又打扰你了。我尝试在 dl3dv 数据集上完成训练过程。在 dl3dv 数据集上训练时，对顶部和底部视图的过度拟合并不明显或常见。然而，当我切换到自定义数据集并训练更少步时，我注意到视图如下： <img alt="Image" width="822.631591796875" height="625.7236938476562" src="https://private-user-images.githubusercontent.com/218775256/530469685-388f2533-0e3f-4a4b-8e13-70f9b63c79cf.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjgwMjcxNTgsIm5iZiI6MTc2ODAyNjg1OCwicGF0aCI6Ii8yMTg3NzUyNTYvNTMwNDY5Njg1LTM4OGYyNTMzLTBlM2YtNGE0Yi04ZTEzLTcwZjliNjNjNzljZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMTEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDExMFQwNjM0MThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mYWU1NmFkNWU2MDY3NzMxYjE1YzQ3MzhkMWY4ZGQxODJmMWEzOTQ2Zjg3NjIzYmEyMjVlOTExZWQ5ZDM4ZmI5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.PEVIbMHqTZQuhmr42bpxe8Tnj0UZObAz5tswz0xwgew"> I want to ask:我想问：
> 
> 1. Is this phenomenon normal?这种现象正常吗？
> 2. My next step is to try loading the anysplat weights or modifying the data loading sampling strategy. Do you have any good suggestions?下一步是尝试加载 anysplat 权重或修改数据加载采样策略。你有什么好建议吗？

**Comment by LiuChunlin-eng** [comment]
> Hello, I am currently attempting to train on a small DL3DV dataset, but I have encountered instability during the training process. May I ask if you have experienced this situation during your training? I was wondering if you are using the original DL3DV dataset, and if there have been any modifications to the parameter configurations or code sections? I look forward to your reply and sincerely appreciate your help.

I previously also tried training on a small DL3DV dataset without making any changes to the parameter settings or code. Here are the results and loss I observed.

<img width="400" height="304" alt="Image" src="https://github.com/user-attachments/assets/0c31585a-e628-4dff-a7ec-b715b1d4a18a" />

<img width="1035" height="590" alt="Image" src="https://github.com/user-attachments/assets/bb23e58c-6e66-4083-84df-ce6f659db9e5" />

**Comment by 20713** [comment]
> Hello, I am currently attempting to train on a small DL3DV dataset, but I have encountered instability during the training process. May I ask if you have experienced this situation during your training? I was wondering if you are using the original DL3DV dataset, and if there have been any modifications to the parameter configurations or code sections? I look forward to your reply and sincerely appreciate your help.
> 
> > Sorry to bother you again. I tried to go through the training process on the dl3dv dataset. When training on the dl3dv dataset, overfitting to the top and bottom views is not obvious or common. However, when I switch to a custom dataset and train for fewer steps, I notice the views are as follows:抱歉又打扰你了。我尝试在 dl3dv 数据集上完成训练过程。在 dl3dv 数据集上训练时，对顶部和底部视图的过度拟合并不明显或常见。然而，当我切换到自定义数据集并训练更少步时，我注意到视图如下： <img alt="Image" width="822.631591796875" height="625.7236938476562" src="https://private-user-images.githubusercontent.com/218775256/530469685-388f2533-0e3f-4a4b-8e13-70f9b63c79cf.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjgwMjcxNTgsIm5iZiI6MTc2ODAyNjg1OCwicGF0aCI6Ii8yMTg3NzUyNTYvNTMwNDY5Njg1LTM4OGYyNTMzLTBlM2YtNGE0Yi04ZTEzLTcwZjliNjNjNzljZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMTEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDExMFQwNjM0MThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mYWU1NmFkNWU2MDY3NzMxYjE1YzQ3MzhkMWY4ZGQxODJmMWEzOTQ2Zjg3NjIzYmEyMjVlOTExZWQ5ZDM4ZmI5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.PEVIbMHqTZQuhmr42bpxe8Tnj0UZObAz5tswz0xwgew"> I want to ask:我想问：
> > 
> > 1. Is this phenomenon normal?这种现象正常吗？
> > 2. My next step is to try loading the anysplat weights or modifying the data loading sampling strategy. Do you have any good suggestions?下一步是尝试加载 anysplat 权重或修改数据加载采样策略。你有什么好建议吗？

 I am also attempting to train on a small DL3DV 1k  dataset, but I encountered instability during the training process. May I ask if you have solved this situation?

<img width="1866" height="645" alt="Image" src="https://github.com/user-attachments/assets/55d5760c-678d-4b4b-b3fa-4902d234680b" />

**Comment by myc634** [comment]
this is weird and never happened before in our experiment. Looks like some issue with view sampling strategy. can u guys start from co3d v2 for training to check whether the training code is bug-free?

**Comment by LiuChunlin-eng** [comment]
> > Hello, I am currently attempting to train on a small DL3DV dataset, but I have encountered instability during the training process. May I ask if you have experienced this situation during your training? I was wondering if you are using the original DL3DV dataset, and if there have been any modifications to the parameter configurations or code sections? I look forward to your reply and sincerely appreciate your help.
> > > Sorry to bother you again. I tried to go through the training process on the dl3dv dataset. When training on the dl3dv dataset, overfitting to the top and bottom views is not obvious or common. However, when I switch to a custom dataset and train for fewer steps, I notice the views are as follows:抱歉又打扰你了。我尝试在 dl3dv 数据集上完成训练过程。在 dl3dv 数据集上训练时，对顶部和底部视图的过度拟合并不明显或常见。然而，当我切换到自定义数据集并训练更少步时，我注意到视图如下： <img alt="图片" width="822.631591796875" height="625.7236938476562" src="https://private-user-images.githubusercontent.com/218775256/530469685-388f2533-0e3f-4a4b-8e13-70f9b63c79cf.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjgwMjcxNTgsIm5iZiI6MTc2ODAyNjg1OCwicGF0aCI6Ii8yMTg3NzUyNTYvNTMwNDY5Njg1LTM4OGYyNTMzLTBlM2YtNGE0Yi04ZTEzLTcwZjliNjNjNzljZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMTEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDExMFQwNjM0MThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mYWU1NmFkNWU2MDY3NzMxYjE1YzQ3MzhkMWY4ZGQxODJmMWEzOTQ2Zjg3NjIzYmEyMjVlOTExZWQ5ZDM4ZmI5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.PEVIbMHqTZQuhmr42bpxe8Tnj0UZObAz5tswz0xwgew"> I want to ask:我想问：
> > > 
> > > 1. Is this phenomenon normal?这种现象正常吗？
> > > 2. My next step is to try loading the anysplat weights or modifying the data loading sampling strategy. Do you have any good suggestions?下一步是尝试加载 anysplat 权重或修改数据加载采样策略。你有什么好建议吗？
> 
> I am also attempting to train on a small DL3DV 1k dataset, but I encountered instability during the training process. May I ask if you have solved this situation?
> 
> <img alt="Image" width="1866" height="645" src="https://private-user-images.githubusercontent.com/70115870/546063135-55d5760c-678d-4b4b-b3fa-4902d234680b.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzI1Mzk5OTMsIm5iZiI6MTc3MjUzOTY5MywicGF0aCI6Ii83MDExNTg3MC81NDYwNjMxMzUtNTVkNTc2MGMtNjc4ZC00YjRiLWIzZmEtNDkwMmQyMzQ2ODBiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAzMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMzAzVDEyMDgxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE0ZDUwY2M0MTJmZTM2YmU1ZDg3MTZhYmI3YjEwZGU0ZjUwZjEwNDNhMjM4YzE3NGNlZWUyZTA3NzI2OGNlNGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Rq2UBRBqeW3Sev5oH50z90L6mlo9vpM_22GLhp3aDbw">

Sorry for not replying to you in time. I have also observed similar phenomena, but I did not continue to study this issue. May I ask if you tried training with a pre-trained model? Has this problem been resolved now?

**Comment by LiuChunlin-eng** [comment]
> this is weird and never happened before in our experiment. Looks like some issue with view sampling strategy. can u guys start from co3d v2 for training to check whether the training code is bug-free?

Yes, it is a code bug issue. It has been identified.
